### PR TITLE
Refactoring SASS

### LINKED ISSUE #11: Sizes has inappropriate format
It's better to use more flexible formats for sizes as percents or em.

### LINKED ISSUE #28: id remove
Need remove ids and apply styles using nesting in sass

### LINKED ISSUE #29: United graphs for activity to one
We have three different graphs for activities properties. May be It's better to united it to one widget where this graphs will be scroll by clicking on side arrow

### PR DESCRIPTION
**Pull request by RayRedGoose**
@JamesRexMiller4 @Garrett-Iannuzzi  @brittanystoroz @khalidwilliams 
I updated toggle for modes and added functionality for switching style. Also I added functionality to dropdown menu filter and for icons buttons.

Screen Recording 2019-10-22 at 11 31 AM

The last thing that I did is refactoring SASS: I changed sizes in px to em values and change CSS custom properties to SASS variables, we still use it but as a values of SASS variables.

You can skip changes inside .svg files, this is adding images. jQuery has approximate code for header elements functionality that will be refactored. CSS changes have priority to review.

This PR resolves #29
This PR resolves #28
This PR resolves #11

### CODE REVIEW

**Comment by RayRedGoose** [comment]
This PR resolves #29

**Comment by RayRedGoose** [comment]
This PR resolves #28

**Comment by RayRedGoose** [comment]
I forget to add that I made functionality for weekly activity information that contains three different graphs. 

Screen Recording 2019-10-22 at 10 24 AM

**Comment by brittanystoroz** [comment]
These design changes are great, you will just want to be very diligent about making sure they work when using a screen reader -- I can imagine some of the toggles and popups when clicking on icons causing accessibility concerns.

**Comment by brittanystoroz** [comment]
There is a lot of COMPLEX Sass stuff in here! Teammates - make sure you thoroughly understand these changes as you review this PR -- if you don't understand how something works, ask. These will all be fair game questions on your eval and it's up to you to know the codebase.

File: src/css/dropdown-menu.scss [hunk 1]
```diff
@@ -1,41 +1,42 @@
...
+    border-radius: 2em;
     cursor: pointer;
     margin: 0;
-    padding: 6px;
+    padding: 0.2em;
     &>p {
-      color: var(--base-color);
-      font-size: 16px;
+      color: $base-color;
+      font-size: 1.1em;
```
Comment (Garrett-Iannuzzi) [inline src/css/dropdown-menu.scss]: @RayRedGoose just going to ask for a few clarifications,  when you use the `&>header` and `&>p` you are selecting the elements within the drop down?

File: src/css/styles.scss [hunk 2]
```diff
@@ -1,12 +1,8 @@
...
-  --section-bcg-color: #484e52;
-  --base-color: #ffffff;
-  --accent-color: #dda0dd;
-}
-
 $base-font: 'Open Sans', sans-serif;
+$bcg-color: var(--bcg-color);
+$section-bcg-color: var(--section-bcg-color);
+$base-color: var(--base-color);
+$accent-color: var(--accent-color);
```
Comment (Garrett-Iannuzzi) [inline src/css/styles.scss]: What does the use pf the `--` here represent? Are you just setting the variable `$bcg-color` to a new variable? Is this different than  ` --bcg-color: #c0dbf5` ? Could we not just save our `$bcg-color =  #c0dbf5` ? Thanks!

File: src/index.html [hunk 3]
```diff
@@ -11,37 +11,32 @@
...
-            <p>All info</p>
-            <p>Today</p>
-            <p>Week</p>
-            <p>Average</p>
-          </div>
-        </section>
-      </div>
+      <form class="toggle">
+      	<input type="checkbox" id="toggle-switch" />
+      	<label for="toggle-switch"><span class="screen-reader-text">Toggle Color Scheme</span></label>
```
Comment (Garrett-Iannuzzi) [inline src/index.html]: Is this span something we could change to a button for accessibility?

File: src/scripts.js [hunk 4]
```diff
@@ -438,3 +442,58 @@ var stepTrend = new Chart(ctx, {
...
+  const $widget = $(this).data('type');
+  const $block = $('#user-admin-info');
+  if ($block.data('type') === $widget) {
+    $block.data('type', '').hide();
+  } else {
+    $block.show().children(`.${$widget}-inputs`).show().siblings().hide();
+    $block.data('type', $widget);
+  }
+});
+
```
Comment (Garrett-Iannuzzi) [inline src/scripts.js]: Could you walk me through what is happening here? You are making a jQuery obj out of the data('type')? This function is responsible for toggling the drop down?
I have seen it in action and it looks great but just need a little clarification. Thanks!

File: src/css/styles.scss [hunk 5]
```diff
@@ -51,65 +51,80 @@ article {
   break-inside: avoid;
```
Comment (JamesRexMiller4) [inline src/css/styles.scss]: I have never heard of break-inside: avoid, might need you to explain this later @RayRedGoose

File: src/index.html (hunk 3 above)
Comment (JamesRexMiller4) [reply src/index.html]: That's a good point @Garrett-Iannuzzi, something we should consider. If we lose some of the aesthetics by switching to a button we could probably add an event listener to the span that fires a onKeyDown event, and only fires if the keycode === 13. Should circle back to this as we design the input forms functionality/accessibility for POST network requests

File: src/index.html [hunk 6]
```diff
@@ -11,37 +11,32 @@
...
-        <li><img src="../images/water.svg" alt="add water icon">
+      <ul class="icons">
+        <li><img class="icon" tabindex="4" src="../images/water-lightmode.svg" alt="add water icon" data-type="water">
         </li>
-        <li><img src="../images/sleep-add.svg" alt="add sleep icon"></li>
-        <li><img src="../images/run.svg" alt="add activity icon"></li>
-        <li><img src="../images/avatar.svg" alt="user icon"></li>
+        <li><img class="icon" tabindex="3" src="../images/sleep-lightmode.svg" alt="add sleep icon" data-type="sleep"></li>
+        <li><img class="icon" tabindex="2" src="../images/activities-lightmode.svg" alt="add activity icon" data-type="activities"></li>
+        <li><img class="icon" tabindex="1" src="../images/user-lightmode.svg" alt="user icon" data-type="user"></li>
```
Comment (JamesRexMiller4) [inline src/index.html]: Nice job adding the tab index for the images!

File: src/scripts.js (hunk 4 above)
Comment (JamesRexMiller4) [reply src/scripts.js]: Second that. Pretty sure this is for the user to input new data by clicking on one of the icons in the right corner of the header. So as we click one child will appear and we hide all the siblings. If the user mouses out of the form field then the form hides again.

File: src/scripts.js [hunk 7]
```diff
@@ -438,3 +442,58 @@ var stepTrend = new Chart(ctx, {
...
+$('.dropdown header').on('click', function() {
+  $('.dropdown div').toggle();
+});
+
+$('.dropdown div p').on('click', function() {
+  $('.dropdown header p').text($(this).text());
+  $('.dropdown input').val($(this).text());
+  $(this).parent().hide();
+});
```
Comment (JamesRexMiller4) [inline src/scripts.js]: Strong use of this being used which is dope, just want to make sure I understand what this is referring in these contexts. On line 477, guessing this is changing the name of the form to reflect as we click across the icons. Nice work making it dynamic and utilizing this in your favor

**Pull request merged**